inc/extras.php

```diff
@@ -685,3 +685,40 @@ function astra_can_remove_elementor_toc_margin_space() {
 	$astra_settings['remove-elementor-toc-margin-css'] = isset( $astra_settings['remove-elementor-toc-margin-css'] ) ? false : true;
 	return apply_filters( 'astra_remove_elementor_toc_margin', $astra_settings['remove-elementor-toc-margin-css'] );
 }
+
+/**
+ * Check whether widget specific config, dynamic CSS, preview JS needs to remove or not. Following cases considered while implementing this.
+ *
+ * 1. Is user is from old Astra setup.
+ * 2. Check if user is new but on lesser WordPress 5.8 versions.
+ * 3. User is new with block widget editor.
+ *
+ * @since x.x.x
+ * @return boolean
+ */
+function astra_remove_widget_design_options() {
+	$astra_settings               = get_option( ASTRA_THEME_SETTINGS );
+	$remove_widget_design_options = isset( $astra_settings['remove-widget-design-options'] ) ? false : true;
+
+	// True -> Hide widget sections, False -> Display widget sections.
+	$is_widget_design_sections_hidden = true;
```

Comment:
Instead of introducing a new variable, we can also use `$remove_widget_design_options` and update the flag(true/false) Also below if condition is also not required in that case.

Reply:
we can eliminate first if condition and only 3 `if-elseif-else` conditions will remain. But then for every old user second elseif condition satisfy and it will hide those options. Whereas it needed to be visible anyhow for old users.

Reply:
Okay Can do like this then,
return early and avoid nesting of code.
```suggestion
	if ( ! $remove_widget_design_options ) {
		// For old users we will show widget design options by anyways.
		return apply_filters( 'astra_remove_widget_design_options', $is_widget_design_sections_hidden );
	}
	// Considering the user is new now.
	if ( isset( $astra_settings['remove-widget-design-options'] ) && $astra_settings['remove-widget-design-options'] ) {
		// User was on WP-5.8 lesser version previously and he may update their WordPress to 5.8 in future. So we display the options in this case.
		$is_widget_design_sections_hidden = false;
	} elseif ( astra_has_widgets_block_editor() ) {
		// User is new & having block widgets active. So we will hide those options.
		$is_widget_design_sections_hidden = true;
	} else {
			// Setting up flag because user is on lesser WP versions and may update WP to 5.8.
			astra_update_option( 'remove-widget-design-options', true );
	}
	

	return apply_filters( 'astra_remove_widget_design_options', $is_widget_design_sections_hidden );
```

Reply:
Done 👍